api/v1alpha3/conversion.go

```diff
@@ -76,14 +88,39 @@ func (r *AWSClusterList) ConvertFrom(srcRaw conversion.Hub) error {
 func (r *AWSMachine) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha4.AWSMachine)
 
-	return Convert_v1alpha3_AWSMachine_To_v1alpha4_AWSMachine(r, dst, nil)
+	if err := Convert_v1alpha3_AWSMachine_To_v1alpha4_AWSMachine(r, dst, nil); err != nil {
+		return err
+	}
+
+	restored := &v1alpha4.AWSMachine{}
+	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
+		return err
+	}
+
+	if restored.Spec.RootVolume != nil && restored.Spec.RootVolume.Encrypted == nil {
```

Comment:
```suggestion
	if dst.Spec.RootVolume != nil && dst.Spec.RootVolume.Encrypted == false && restored.Spec.RootVolume != nil && restored.Spec.RootVolume.Encrypted == nil {
```

Although this shouldn't happen, we shouldn't change true back to false.